seonjeong/app.js

```diff
@@ -51,6 +51,145 @@ app.post("/users", async (req, res) => {
   return res.status(201).json({ message: "successfully created" });
 });
 
+app.post("/posts", async (req, res) => {
+  const { title, content, imageUrl, userId } = req.body;
+
+  await appDataSource.query(
+    `INSERT INTO posts (
+      title,
+      content,
+      image_url,
+      user_id
+    ) VALUES (?, ?, ?, ?);`,
+    [title, content, imageUrl, userId]
+  );
+  return res.status(201).json({ message: "postCreated" });
+});
+//동기: 융통성x, 비동기: 융통성o
+//async + await
+
+app.get("/search", async (req, res) => {
+  const result = await appDataSource.query(
+    `SELECT
+    users.id as userId,
+    users.profile_image as userProfileImage,
+    posts.id as postingId,
+    posts.image_url as postingImageUrl,
+    posts.content as postingContent
+    FROM users INNER JOIN posts
+    ON users.id = posts.user_id
+    `
+  );
+
+  return res.status(200).json({ data: result });
+});
+
+app.get("users/userId/posts", async (req, res) => {
```

Comment:
해당 api 테스트가 된걸까요? params를 사용하기 위해서는 사용하고 싶은 parameter에 :을 붙여야 합니다.
```suggestion
app.get("users/:userId/posts", async (req, res) => {
```